cmd/installer/cli/upgrade.go

```diff
@@ -679,3 +679,127 @@ func validateIsReleaseUpgradable(ctx context.Context, upgradeConfig *upgradeConf
 
 	return nil
 }
+
+// runMigrationAPI starts the API server in migration mode for kURL to EC migration.
+// TODO: This is a minimal implementation. Future enhancements needed:
+// - Read TLS certificates from kURL cluster
+// - Add proper error handling and cleanup
+func runMigrationAPI(
```

Comment:
how about moving the common logic / functionality to a common function and use it in both `runMigrationAPI` and `runManagerExperienceUpgrade`?